### DISCUSSION TITLE
large video files?

### DISCUSSION THREAD

**Discussion by pmavros** [jspsych/jsPsych#2471] [Q&A]
Hello everyone,

We want to run an online experiment where we show participants video-clips of about 30 seconds and then collect their responses. I have figured out how to implenent this in jsPsych, but have the following question:

Our videos are 1080p HD, so they are anbout 70-90MB each. I am wondering/worrying if this is going to be an issue, with slow loading times etc. Does anyone have experience if it is better to use a service (e.g. vimeo, based on other discussions here) or host in the same place as the task (pavlovia, jatos or cognition)?

Any advice between pavlovia, jatos and cognition is most welcome too!

thanks in advance,
Panos

**Comment by jodeleeuw** [comment]
Hi @pmavros,

I think the video plugins will handle 70-90MB files just fine, but I'd probably avoid trying to preload the video content unless absolutely essential. The video plugins should be able to play the video while downloading the files, reducing the wait time for the participants.

Hosting the video files on a service like vimeo or YouTube would probably improve the download speeds. Those services almost certainly have a better delivery infrastructure than whatever server is hosting the experiment, and those services may use compression techniques that reduce the amount of data that needs to be sent.

**Comment by pmavros** [comment]
Hi @jodeleeuw 

thanks for the quick response! 

I looked into vimeo hosting for the same reason. Looking at the jsPsych documentation and some discussion here it seems if I want to play a video hosted externally, i have to use the html~ plugin with <video src="..."></video>, but then i lose all the controls on the video is that correct?

could i still use the video plugin pointing to the video link instead of my local/server file?

thanks again!

**Reply by pmavros** [reply]
Well it seems vimeo can also serve videos as files https://vimeo.zendesk.com/hc/en-us/articles/224823567-Direct-links-to-video-files in case this is the only option

**Reply by jodeleeuw** [reply]
If you use the `html-` plugins then you could presumably control some of the video stuff with the markup on the embedded video player. I imagine that there are parameters you can set in the HTML that will affect playback, but I haven't tried this myself.

**Reply by sophiehoyer** [reply]
Hi @pmavros !

I am running a somewhat similar experiment set up where I am showing a very large number of smaller videos (300 3 second videos, each ~4.5 MB). I wanted to ask: what did you end up implementing for this experiment?

Thanks!
Sophie

**Comment by pmavros** [comment]
Hi @sophiehoyer, 

Thanks for the question. 

tldr: vimeo + cognition.run + prolific.

Our video stimuli at the time were around 70-100MB each. We recruited participants from prolific across the globe, so we opted not to put them in our own server, hoping that a commercial provider would provide better streaming across countries. In the end we uploaded them to vimeo, and change the default settings to private so you only see the video if you have the link. Vimeo has an api so it was convenient to query all the links etc. After data collection, we tried to estimate quality of playback by comparing how long each video actually played for versus it's duration – e.g. due to streaming issues the effective duration was 10-20 seconds longer than the actual duration. If the delay was 'too much' we excluded the stimuli from analysis. With small file sizes that you describe, hopefully that won't be an issue, but it is worth thinking how you can validate playback quality. 

Since then, cognition.run now has a paid version where you can host larger stimuli, but I have not tested it.

Let me know if you have more questions!
Cheers, 
Panos
